eudemo/eudemo/maintenance/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+        self.off = offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the castellated component.
+        """
+        # TODO: Implement corner radii using PR #1992
+        self.r_rad = self.params.cst_r_corner.value
+        self.n = self.params.n_components.value
+
+        xyz_solid = self.build_castellations(
+            self.face, self.vec, self.length, self.off, self.cst
+        )
+        xz_plane = BluemiraPlane(axis=(0, 1, 0))
+        xy_plane = BluemiraPlane(axis=(0, 0, 1))
+        xz_slice = BluemiraFace(slice_shape(xyz_solid, xz_plane))
+        xy_slice = BluemiraFace(slice_shape(xyz_solid, xy_plane))
+
+        return self.component_tree(
+            xz=[self.build_xz(xz_slice)],
+            xy=self.build_xy(xy_slice),
+            xyz=self.build_xyz(xyz_solid),
+        )
+
+    def build_xz(self, xz: BluemiraFace) -> PhysicalComponent:
+        """
+        Build the xz representation of the castellated component
+        """
+        body = PhysicalComponent(self.NAME, xz)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+        return body
+
+    def build_xy(self, xy: BluemiraFace, n: int = 10) -> List[PhysicalComponent]:
+        """
+        Build the xy representation of the castellated component
+        """
+        body = PhysicalComponent(self.NAME, xy)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+        return circular_pattern_component(body, n_children=n)
+
+    def build_xyz(self, xyz: BluemiraSolid, n: int = 10) -> List[PhysicalComponent]:
+        """
+        Build the xyz representation of the castellated component
+        """
+        cst_shape = PhysicalComponent(self.NAME, xyz)
+        cst_shape.display_cad_options.color = BLUE_PALETTE["VV"][0]
+        return circular_pattern_component(cst_shape, n_children=n)
+
+    def build_castellations(
+        self,
+        face: BluemiraFace,
+        vec: tuple,
+        length: float,
+        distances: List[float],
+        offsets: List[float],
+    ) -> BluemiraSolid:
+        """
+        Returns BluemiraSolid for a BluemiraFace castellated along a given vector
+
+        Parameters
+        ----------
+        wire: BluemiraWire
+            profile boundary to be castellated
+        vec: tuple (x,y,z)
+            unit vector along which to extrude
+        length: float
+            total length of castellated BluemiraSolid in vec direction
+        distances: List[float]
+            castellation positions along vec (relative to starting wire)
+        offsets: List[float]
+            castellations offsets for each position
+        """
+        base = face
+        sections = []
+
+        # Normalise vec
+        vec_mag = np.linalg.norm(vec)
+        if vec_mag != 1.0:
+            vec /= vec_mag
+
+        parameter_array = list(zip(distances, offsets))
+        parameter_array.append((length, 1.0))
+        _prev_dist = 0
```

Comment:
This like it could be done in a cuter way. The 1.0 at the end is just to enable your zipped loop to have a value with which to offset in the final iteration?

If it must be so, please use a final offset value of 0.0, as this won't do much unnecessary computation (at least compared to an offset of 1.0).